lib/logstash/util/accessors.rb

```diff
@@ -48,7 +48,9 @@ def strict_set(accessor, value)
 
     def del(accessor)
       target, key = lookup(accessor)
-      target.is_a?(Array) ? target.delete_at(key.to_i) : target.delete(key)
+      unless target.nil?
```

Comment:
Full confession, I don't like that nil is what we return for an accessor that doesn't exist, but that's more of a historical artifact than anything else. Still, I think the `return if target.nil?` is a good option, here.